rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -1021,6 +1030,10 @@ stream {
             {{ end }}
 
             rewrite_by_lua_block {
+                -- for the plugins enabled for this location
+                -- in this phase, run the like following
```

Comment:
one can argue that the plugin itself knows better where it should be run - so maybe make this part of the plugin?

but on the other side there can be plugins that can be run in any phase maybe? and you'd wanna run it only in certain phase for your app

Reply:
another question is do we wanna have two different apps configure the same plugin in different phases?

probably not

Reply:
maybe it's better to have plugins to define their public API based on phase name
and have plugin runner to check whether there's a function defined for the given phase in the given plugin and if so then run it.

Reply:
I think it makes sense to declare the phase a plugin must be run into the plugin definition itself.
Wither by using an exported array with phase list or having a separate function for each phase.